cmake/modules/Hexagon.cmake

```diff
@@ -116,7 +116,7 @@ function(add_hexagon_wrapper_paths)
   link_directories("${HEXAGON_TOOLCHAIN}/lib/iss")
 endfunction()
 
-if(BUILD_FOR_HEXAGON OR USE_HEXAGON_RPC)
+if(BUILD_FOR_HEXAGON OR ("${USE_HEXAGON_RPC}" STREQUAL "HEXAGON"))
```

Comment:
When would `BUILD_FOR_HEXAGON` be false, but `USE_HEXAGON_RPC` be `"HEXAGON"`?  It seems like the latter implies the former...

Reply:
BUILD_FOR_HEXAGON is false for the runtime android and x86 build targeted by the hexagon_api, and USE_HEXAGON_RPC is true for these cases. Thus I want to tighten the requirement so that hexagon sources are not brought in to the tvm_runtime builds (initiated by the hexagon_api cmake) when we aren't building for hexagon.

Reply:
What I mean is that it seems that `(${USE_HEXAGON_RPC} == "HEXAGON")  =>  BUILD_FOR_HEXAGON`.  Unless I'm missing something, you can just remove `USE_HEXAGON_RPC` from x86/android builds.

Reply:
I tested with `if(BUILD_FOR_HEXAGON)` and it appears to work at least for PR #11653 which depends on this PR to pass CI.

Reply:
I am getting some strange results if I fail to set `USE_HEXAGON_RPC` for the x86/Android builds.  For Android, there is a linker error.  For x86, some sort of runtime simulation error.  I did not spend time to debug these errors.

However, I can revert the changes to `hexagon_api/CMakeLists.txt` and just set `USE_HEXAGON_RPC=ON` for all three builds while checking for `if(BUILD_FOR_HEXAGON)` in Hexagon.cmake and it appears to work.

Reply:
We may be able to "remove `USE_HEXAGON_RPC` from x86/android builds" in `hexagon_api/CMakeLists.txt` as @kparzysz-quic but I am getting some strange results when doing so.  I reverted the changes to that file in this PR.  

Changing to `if(BUILD_FOR_HEXAGON)` and removing the API check works and accomplishes the original intent of the PR.